### PR TITLE
SSH tool

### PR DESCRIPTION
**Pull request by jim-minter**
Tool to SSH into clusters to diagnose them.

Status: completely untested.

### CODE REVIEW

**Comment by jim-minter** [comment]
@asalkeld @mjudeikis please can you test and review this as a priority?

File: hack/sshtool/azure.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+			Name: to.StringPtr(fmt.Sprintf("ssh-%d", i)),
+		})
+
+		*lb.LoadBalancingRules = append(*lb.LoadBalancingRules, mgmtnetwork.LoadBalancingRule{
+			LoadBalancingRulePropertiesFormat: &mgmtnetwork.LoadBalancingRulePropertiesFormat{
+				FrontendIPConfiguration: &mgmtnetwork.SubResource{
+					ID: (*lb.FrontendIPConfigurations)[0].ID,
+				},
+				BackendAddressPool: &mgmtnetwork.SubResource{
+					ID: to.StringPtr(fmt.Sprintf("%s/probes/backendAddressPools/ssh-%d", *lb.ID, i)),
```
Comment (asalkeld) [inline hack/sshtool/azure.go:144]:
```suggestion
					ID: to.StringPtr(fmt.Sprintf("%s/backendAddressPools/ssh-%d", *lb.ID, i)),
```

File: hack/sshtool/azure.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	for _, c := range *lb.LoadBalancingRules {
+		if !strings.HasPrefix(*c.Name, "ssh-") {
+			loadBalancingRules = append(loadBalancingRules, c)
+		}
+	}
+	*lb.LoadBalancingRules = loadBalancingRules
+
+	probes := make([]mgmtnetwork.Probe, 0, len(*lb.Probes))
+	for _, p := range *lb.Probes {
+		if !strings.HasPrefix(*p.Name, "ssh-") {
```
Comment (asalkeld) [inline hack/sshtool/azure.go:123]:
```suggestion
		if !strings.EqualFold(*p.Name, "ssh") {
```

File: hack/sshtool/azure.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	for _, p := range *lb.BackendAddressPools {
+		if !strings.HasPrefix(*p.Name, "ssh-") {
+			backendAddressPools = append(backendAddressPools, p)
+		}
+	}
+	*lb.BackendAddressPools = backendAddressPools
+
+	loadBalancingRules := make([]mgmtnetwork.LoadBalancingRule, 0, len(*lb.LoadBalancingRules))
+	for _, c := range *lb.LoadBalancingRules {
+		if !strings.HasPrefix(*c.Name, "ssh-") {
```
Comment (asalkeld) [inline hack/sshtool/azure.go:115]: not sure why i got nil names..
```suggestion
		if p.Name != nil &&!strings.HasPrefix(*c.Name, "ssh-") {
```

File: hack/sshtool/azure.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	}
+
+	(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = &backendAddressPools
+}
+
+func enableNIC(nic *mgmtnetwork.Interface, lb *mgmtnetwork.LoadBalancer, i int) {
+	disableNIC(nic)
+
+	*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
+		ID: to.StringPtr(fmt.Sprintf("%s/probes/backendAddressPools/ssh-%d", *lb.ID, i)),
```
Comment (asalkeld) [inline hack/sshtool/azure.go:100]:
```suggestion
		ID: to.StringPtr(fmt.Sprintf("%s/backendAddressPools/ssh-%d", *lb.ID, i)),
```

**Comment by jim-minter** [comment]
thanks @asalkeld, please can you copy my branch, add commit(s) to it, and open a PR to replace this one?

**Comment by asalkeld** [comment]
> thanks @asalkeld, please can you copy my branch, add commit(s) to it, and open a PR to replace this one?

sure, will do

**Comment by asalkeld** [comment]
closing this for https://github.com/Azure/ARO-RP/pull/736

**Pull request closed without merging**